core/src/main/java/org/jdbi/v3/core/internal/enums/ByName.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.internal.enums;
+
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.jdbi.v3.core.EnumByName;
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.Arguments;
+import org.jdbi.v3.core.mapper.ColumnMapper;
+import org.jdbi.v3.core.result.UnableToProduceResultException;
+import org.jdbi.v3.core.statement.StatementContext;
+
+@EnumByName
+public class ByName implements EnumStrategy {
+    private static final ByName INSTANCE = new ByName();
+
+    public static ByName singleton() {
```

Comment:
I left these in from the original rework but I think we can reasonably forget about caching the instances